### PR TITLE
Update API Spec for New Code Tables

### LINKED ISSUE #153: Update api-doc-security-filter.ts to use new code tables
**Describe the Task**
Update api-doc-security-filter.ts to use new code tables. Updates or deletes need to happen to these files in order to make the new code tables work with the drop-down logic. 

**Acceptance Criteria**
- [x] Update api-doc-security-filter.ts to use new code tables
- [x] Updates to code-queries.ts and code-utils.ts
- [x] Update all of the 'x-enum-code' fields in api-doc.json
- [ ] update api doc to account for new code tables

### PR DESCRIPTION
**Pull request by sdevalapurkar**
## Overview

This PR updates the API specification to account for the code tables and code header names as they appear in the database to maintain consistency across the application.

The following fields have been added for specific cases. They are all mandatory but `treatment_issues_code` has a default "none" as mandatory.

- `pest_management_plan` -> chemical plants
- `treatment_issues_code` -> all plant treatments
- `invasive_plant_code` -> all plant activities
- `invasive_animal_code` -> all animal activities
- `liquid_herbicide_code` -> for plants

Note: `herbicide_name` field has been changed to `liquid_herbicide_code` enum as per discussion with Shasko. The granule herbicide is being left out for now, only using liquid.

## Screenshots

Screenshot 2020-11-25 155319

## Leaving out for now

- app role code
- granule herbicide code
- observation geometry code

## Ticket Reference

#153

### CODE REVIEW

File: api/src/openapi/api-doc.json [hunk 1]
```diff
@@ -593,18 +600,18 @@
...
               "x-enum-code-category-name": "invasives",
-              "x-enum-code-header-name": "observation_aspect_code",
+              "x-enum-code-header-name": "aspect_code",
               "x-enum-code-name": "code_name",
               "x-enum-code-text": "code_description",
               "x-enum-code-sort-order": "code_sort_order"
             }
           },
-          "proposed_action_code": {
+          "proposed_treatment_code": {
```
Comment (sdevalapurkar) [inline api/src/openapi/api-doc.json:587]: @NickPhura this one I am a bit unsure about but I think it made sense and that these two were referring to the same field

Comment (micheal-w-wells) [reply api/src/openapi/api-doc.json:587]: what kind of values are in that table/code group?

Comment (sdevalapurkar) [reply api/src/openapi/api-doc.json:587]: examples are:
```
chemical treatment
chemical monitoring
biological treatment
biological monitoring
biological collection
biological dispersal survey
```

**Comment by NickPhura** [comment]
Regarding the `app role code` that is just used for the user roles on the backend, so it won't be part of the schema like the others.  Its the only code set listed under the `system` category.

Can't speak to the specifics of those few enums you had questions about, but everything else looks good.

**Review by NickPhura — DISMISSED** [review]

**Review by NickPhura — APPROVED** [review]

**Pull request merged**